src/lib/homestore/shard_manager.cpp

```diff
@@ -61,183 +62,119 @@ ShardManager::Result< ShardInfo > HSHomeObject::_create_shard(pg_id pg_owner, ui
         LOGWARN("failed to create shard with non-exist pg [{}]", pg_owner);
         return folly::makeUnexpected(ShardError::UNKNOWN_PG);
     }
-    // TODO: will update to ReplDev when ReplDev on HomeStore is ready;
-    auto replica_set_var = _repl_svc->get_replica_set(fmt::format("{}", pg_owner));
-    if (std::holds_alternative< home_replication::ReplServiceError >(replica_set_var)) {
+
+    homestore::ReplicationService* replication_service =
+        (homestore::ReplicationService*)(&homestore::HomeStore::instance()->repl_service());
+    auto repl_dev = replication_service->get_replica_dev(pg.value().repl_dev_uuid);
+    if (!bool(repl_dev)) {
         LOGWARN("failed to get replica set instance for pg [{}]", pg_owner);
         return folly::makeUnexpected(ShardError::UNKNOWN_PG);
     }
 
-    auto replica_set = std::get< home_replication::rs_ptr_t >(replica_set_var);
     auto new_shard_id = generate_new_shard_id(pg_owner);
     auto create_time = get_current_timestamp();
     auto shard_info =
         ShardInfo(new_shard_id, pg_owner, ShardInfo::State::OPEN, create_time, create_time, size_bytes, size_bytes, 0);
     std::string create_shard_message = serialize_shard(Shard(shard_info));
-    // preapre msg header;
-    const uint32_t needed_size = sizeof(ReplicationMessageHeader) + create_shard_message.size();
-    auto buf = nuraft::buffer::alloc(needed_size);
-    uint8_t* raw_ptr = buf->data_begin();
-    ReplicationMessageHeader* header = new (raw_ptr) ReplicationMessageHeader();
-    header->message_type = ReplicationMessageType::SHARD_MESSAGE;
-    header->payload_size = create_shard_message.size();
-    header->payload_crc =
-        crc32_ieee(init_crc32, r_cast< const uint8_t* >(create_shard_message.c_str()), create_shard_message.size());
-    header->header_crc = header->calculate_crc();
-    raw_ptr += sizeof(ReplicationMessageHeader);
-    std::memcpy(raw_ptr, create_shard_message.c_str(), create_shard_message.size());
 
+    // round up msg size to homestore datasvc blk size
+    const auto datasvc_blk_size = homestore::HomeStore::instance()->data_service().get_blk_size();
+    auto msg_size = create_shard_message.size();
+    if (msg_size % datasvc_blk_size != 0) { msg_size = (msg_size / datasvc_blk_size + 1) * datasvc_blk_size; }
```

Comment:
sisl::round_up(msg_size, homestore::HomeStore::instance()->data_service().get_blk_size());

Reply:
good suggestion and make it more clear.